doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,574 @@
+  * AS can be either the AS number seperated by underscores (e.g. ``ff00_0_110`` or a wildcard (``0``).
+  * IGIF can be either the ingress interface number (e.g. ``42``), or a wildcard (``0``).
+  * EGIF can be either the egress interface number (e.g. ``41``, or a wildcard (``0``).
+  * POLICY can be either the policy to apply, where a local policy is denoted as ``L`` + the policy identifier (e.g. ``L100``) and a global policy is denoted by ``G`` + the policy identifier (e.g. ``G100``), a wildcard (``0``), or a rejection ``REJECT``. Rejection means that this path should not be chosen. 
+
+  When used in a query, the identifier evaluates to true when at least a single hop in the path matches the identifier.
+
+* **Concatenations**
+
+  Multiple identifiers can be combined by using a concatenation (and/or parentheses). Concatenations are created by the ``+`` symbol. 
+
+  Example:
+
+  ``(1-0#0,0@G300 + 1-0#0,0@G200)`` applies both policy G300 and policy G200.
+
+  When used in a query, a concatenation evaluates to true when *all* identifiers in the concatenation also evaluate to true.
+
+
+* **Queries**
+
+  You can query for the existence of a specific hop and/or policy through a query. Queries are structured as follows: ``{ QUERY_EXPRESSION ? EXPRESSION_IF_TRUE : EXPRESSION_IF_FALSE}``. The query expression is evaluated, and if an identifier matches with a specific hop, the ``expression_if_true`` branch is applied. If no matches can be found in the path, the ``expression_if_false`` branch is applied. Identifiers in the query expression are not applied, e.g. if a query expression queries for a specific policy, the specific policy is not applied to the hops it matches, unless the same expression is also given under the ``expression_if_true`` branch.
+
+  Example:
+
+  With the path
+  ``1-ff00:0:109#0,5@() 1-ff00:0:110#4,1@(G100, G200) -> 1-ff00:0:111#2,0@(G200, G300)``
+
+  When an expression queries for ``1-0#0,0@G200`` using ``{1-0#0,0@G200 ? 1-0#0,0@G300 : 1-0#0,0@REJECT}``, the policies that are applied to the hops are only policy G300 for the last hop. To also apply policy G200, the query has to be structured as ``{1-0#0,0@G200 ? (1-0#0,0@G300 + 1-0#0,0@G200) : 1-0#0,0@REJECT}``.
+
+  When a query is used within another query, the query_expression is first used to determine which branch is used for the result. If the query would apply the ``expression_if_true`` branch, the result of the query is the evaluation of the ``expression_if_true`` branch. The same applies for the ``expression_if_false`` branch. 
+
+**Evaluation Order**
+The language is evaluated left to right, for each hop only a single policy can be applied. The first identifier match applies the policy, so the order of the query is important. 
+
+Example:
+
+``(0-0#0,0@REJECT + 1-0#0,0@00)``
+
+Will reject all paths, whereas
+
+``(1-0#0,0@0 + 0-0#0,0@REJECT)``
+
+Will reject all paths that are not within ISD 1.
+
+**Possible Extensions**
```

Comment:
Please move to "Considerations for future work"

Reply:
Done.